be/src/vec/functions/array/function_array_cum_sum.cpp

```diff
@@ -0,0 +1,255 @@
+        bool is_nullable = nested_type->is_nullable();
+
+        auto non_null_nested_type = remove_nullable(nested_type);
+        auto scale = get_decimal_scale(*non_null_nested_type);
+        WhichDataType which(non_null_nested_type);
+        //return type is promoted to prevent result overflow
+        //like: input is int32 ---> return type will be int64
+        DataTypePtr return_type = nullptr;
+
+        // same cast logic as array_sum
+        if (which.is_uint8() || which.is_int8() || which.is_uint16() || which.is_int16() ||
+            which.is_uint32() || which.is_int32() || which.is_int64() || which.is_uint64()) {
+            return_type = std::make_shared<DataTypeInt64>();
+        } else if (which.is_int128()) {
+            return_type = std::make_shared<DataTypeInt128>();
+        } else if (which.is_float32() || which.is_float64()) {
+            return_type = std::make_shared<DataTypeFloat64>();
+        } else if (which.is_decimal128() && !which.is_decimal128i()) {
+            // decimalv2
+            return_type = std::make_shared<DataTypeDecimal<Decimal128>>(
+                    DataTypeDecimal<Decimal128>::max_precision(), scale);
+        } else if (which.is_decimal()) {
+            return_type = std::make_shared<DataTypeDecimal<Decimal128I>>(
+                    DataTypeDecimal<Decimal128I>::max_precision(), scale);
+        }
+        if (return_type) {
+            return std::make_shared<DataTypeArray>(is_nullable ? make_nullable(return_type)
+                                                               : return_type);
+        } else {
+            LOG(FATAL) << "Function of " << name
+                       << " return type get wrong: and input argument is: "
+                       << arguments[0]->get_name();
+        }
+
+        return nullptr;
+    }
+
+    Status execute_impl(FunctionContext* context, Block& block, const ColumnNumbers& arguments,
+                        const size_t result, size_t input_rows_count) override {
+        auto src_arg = block.get_by_position(arguments[0]);
+        ColumnPtr src_column = src_arg.column->convert_to_full_column_if_const();
+
+        const auto& src_column_array = check_and_get_column<ColumnArray>(src_column.get());
+        if (!src_column_array) {
+            return Status::RuntimeError(
+                    fmt::format("unsupported types for function {}({})", get_name(),
+                                block.get_by_position(arguments[0]).type->get_name()));
+        }
+
+        const auto& src_offsets = src_column_array->get_offsets();
+        const auto* src_nested_column = &src_column_array->get_data();
+        DCHECK(src_nested_column != nullptr);
+
+        // get src nested column
+        auto src_nested_type = assert_cast<const DataTypeArray&>(*src_arg.type).get_nested_type();
+
+        // get null map
+        const NullMapType* src_null_map = nullptr;
+        if (src_nested_column->is_nullable()) {
```

Comment:
could remove this check, now the nested column of array  must be nullable column in doris

Reply:
Ok, I remove the check, and modify the source code